Missing steps in installation.

The instructions on the [SocketStrream](http://socketstream.org/) website do not work. The installation process is broken.
````
npm install -g socketstream 
socketstream new my_app 
cd my_app 
npm install 
npm start
````
1. `npm install -g socketstream ` Will not install devDependencies which are needed to run the app. 
2. Thus, will brake `socketstream new my_app` and getting error `Cannot find module 'commander'`. After running `cd /path/to/npm/global/socketstream/path` and `npm install` we can then run `socketstream new my_app` successfully.
3. After running `npm install` and `npm start` we get another error `Error: Cannot find module "connect" in socketstream or project`. Which, I asume means that our project's dependencies for socketstream are missing and that connect is not on our dependencies package.json. 
4. After going to our project's `socketstrteam` dependency folder and running `npm install` and back to our project's root and running `npm install connect --save` will address the issue, but I do not know if the dependency should be for socketstream or hour project's.

I guess we could add a post install script on the socketstream package.json but I am new to making PR, if I figure it out, I will add the pull request.

Thanks.

@paulbjensen Can you please reopen this issue as its partially solved? I have installed the develop branch as my global `socketstream` module. I accomplished this by cloning the develop git branch to my computer and then using that folder to install the module with npm: `npm install -g "/path/to/git/folder/"`

However I still get the same error for installing new socketstream app (__original issue [# 1](https://github.com/socketstream/socketstream/blob/develop/bin/socketstream#L4)__) :

Please note that I have set a custom path for my npm installation.

```
socketstream new ss-app2
module.js:338
    throw err;
    ^

Error: Cannot find module 'commander'
    at Function.Module._resolveFilename (module.js:336:15)
    at Function.Module._load (module.js:286:25)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at Object.<anonymous> (C:\dev\npm\node_modules\socketstream\bin\socketstream:4:15)
    at Module._compile (module.js:434:26)
    at Object.Module._extensions..js (module.js:452:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Function.Module.runMain (module.js:475:10)
```

After getting this error, I did the following:

```
cd C:\dev\npm\node_modules\socketstream
npm install
```
The output of that command was as follows:
```
npm WARN deprecated lodash@0.9.2: lodash@<2.0.0 is no longer maintained. Upgrade to lodash@^3.0.0
npm WARN optional dep failed, continuing fsevents@1.0.6
npm WARN deprecated lodash@1.0.2: lodash@<2.0.0 is no longer maintained. Upgrade to lodash@^3.0.0

> spawn-sync@1.0.13 postinstall C:\dev\npm\node_modules\socketstream\node_modules\pre-commit\node_modules\cross-spawn\node_modules\spawn-sync
> node postinstall


> pre-commit@1.1.2 install C:\dev\npm\node_modules\socketstream\node_modules\pre-commit
> node install.js

npm WARN engine hawk@0.10.2: wanted: {"node":"0.8.x"} (current: {"node":"4.1.1","npm":"2.14.4"})
npm WARN engine sntp@0.1.4: wanted: {"node":"0.8.x"} (current: {"node":"4.1.1","npm":"2.14.4"})
npm WARN engine boom@0.3.8: wanted: {"node":"0.8.x"} (current: {"node":"4.1.1","npm":"2.14.4"})
npm WARN engine cryptiles@0.1.3: wanted: {"node":"0.8.x"} (current: {"node":"4.1.1","npm":"2.14.4"})
npm WARN engine hoek@0.7.6: wanted: {"node":"0.8.x"} (current: {"node":"4.1.1","npm":"2.14.4"})
sinon-chai@2.8.0 node_modules\sinon-chai

connect-livereload@0.5.4 node_modules\connect-livereload

notes@0.0.4 node_modules\notes

gently@0.10.0 node_modules\gently

lolex@1.4.0 node_modules\lolex

semver@5.0.3 node_modules\semver

commander@2.8.1 node_modules\commander
└── graceful-readlink@1.0.1

cookie-parser@1.4.0 node_modules\cookie-parser
├── cookie-signature@1.0.6
└── cookie@0.2.2

grunt-contrib-clean@0.6.0 node_modules\grunt-contrib-clean
└── rimraf@2.2.8

shelljs@0.5.3 node_modules\shelljs

express-session@1.12.1 node_modules\express-session
├── on-headers@1.0.1
├── cookie-signature@1.0.6
├── depd@1.1.0
├── crc@3.3.0
└── uid-safe@2.0.0 (base64-url@1.2.1)

should@7.1.1 node_modules\should
├── should-type@0.2.0
├── should-equal@0.5.0
└── should-format@0.3.1

npm-dview@2.0.0 node_modules\npm-dview
├── progress@1.1.8
├── supports-color@3.1.2 (has-flag@1.0.0)
└── cli-table@0.3.1 (colors@1.0.3)

grunt-conventional-changelog@4.1.0 node_modules\grunt-conventional-changelog
├── q@1.4.1
├── plur@2.1.2 (irregular-plurals@1.1.0)
├── chalk@1.1.1 (escape-string-regexp@1.0.4, ansi-styles@2.1.0, supports-color@2.0.0, strip-ansi@3.0.0, has-ansi@2.0.0)
└── concat-stream@1.5.1 (inherits@2.0.1, typedarray@0.0.6, readable-stream@2.0.5)

chai@3.4.1 node_modules\chai
├── assertion-error@1.0.1
├── type-detect@1.0.0
└── deep-eql@0.1.3 (type-detect@0.1.1)

grunt-ngdocs@0.2.9 node_modules\grunt-ngdocs
├── marked@0.3.5
└── shelljs@0.3.0

grunt-contrib-watch@0.6.1 node_modules\grunt-contrib-watch
├── async@0.2.10
├── lodash@2.4.2
├── tiny-lr-fork@0.0.5 (debug@0.7.4, qs@0.5.6, faye-websocket@0.4.4, noptify@0.0.3)
└── gaze@0.5.2 (globule@0.1.0)

supertest@1.1.0 node_modules\supertest
├── methods@1.1.1
└── superagent@1.3.0 (extend@1.2.1, cookiejar@2.0.1, methods@1.0.1, reduce-component@1.0.1, component-emitter@1.1.2, mime@1.3.4, qs@2.3.3, readable-stream@1.0.27-1, formidable@1.0.14, form-data@0.2.0)

grunt-cli@0.1.13 node_modules\grunt-cli
├── nopt@1.0.10 (abbrev@1.0.7)
├── resolve@0.3.1
└── findup-sync@0.1.3 (lodash@2.4.2, glob@3.2.11)

sinon@1.17.2 node_modules\sinon
├── formatio@1.1.1
├── samsam@1.1.2
├── lolex@1.3.2
└── util@0.10.3 (inherits@2.0.1)

pre-commit@1.1.2 node_modules\pre-commit
├── which@1.1.2 (is-absolute@0.1.7)
└── cross-spawn@2.0.1 (cross-spawn-async@2.1.1, spawn-sync@1.0.13)

grunt@0.4.5 node_modules\grunt
├── dateformat@1.0.2-1.2.3
├── which@1.0.9
├── async@0.1.22
├── getobject@0.1.0
├── colors@0.6.2
├── lodash@0.9.2
├── rimraf@2.2.8
├── hooker@0.2.3
├── grunt-legacy-util@0.2.0
├── exit@0.1.2
├── nopt@1.0.10 (abbrev@1.0.7)
├── coffee-script@1.3.3
├── iconv-lite@0.2.11
├── underscore.string@2.2.1
├── minimatch@0.2.14 (sigmund@1.0.1, lru-cache@2.7.3)
├── glob@3.1.21 (inherits@1.0.2, graceful-fs@1.2.3)
├── grunt-legacy-log@0.1.3 (grunt-legacy-log-utils@0.1.1, lodash@2.4.2, underscore.string@2.3.3)
├── findup-sync@0.1.3 (lodash@2.4.2, glob@3.2.11)
└── js-yaml@2.0.5 (esprima@1.0.4, argparse@0.1.16)

grunt-contrib-connect@0.11.2 node_modules\grunt-contrib-connect
├── opn@1.0.2
├── async@0.9.2
├── serve-static@1.10.0 (escape-html@1.0.2)
├── portscanner@1.0.0 (async@0.1.15)
├── morgan@1.6.1 (basic-auth@1.0.3, on-headers@1.0.1, depd@1.0.1, on-finished@2.3.0)
├── connect@3.4.0 (finalhandler@0.4.0)
└── serve-index@1.7.2 (escape-html@1.0.2, batch@0.5.2, http-errors@1.3.1, mime-types@2.1.8, accepts@1.2.13)

mocha@2.3.4 node_modules\mocha
├── escape-string-regexp@1.0.2
├── commander@2.3.0
├── diff@1.4.0
├── growl@1.8.1
├── supports-color@1.2.0
├── mkdirp@0.5.0 (minimist@0.0.8)
├── jade@0.26.3 (commander@0.6.1, mkdirp@0.3.0)
└── glob@3.2.3 (inherits@2.0.1, graceful-fs@2.0.3, minimatch@0.2.14)

dgeni@0.4.1 node_modules\dgeni
├── canonical-path@0.0.2
├── di@0.0.1
├── q@0.9.7
├── lodash@2.4.2
├── validate.js@0.2.0
├── dependency-graph@0.1.0 (underscore@1.4.4)
├── optimist@0.6.1 (wordwrap@0.0.3, minimist@0.0.10)
└── winston@0.7.3 (cycle@1.0.3, async@0.2.10, stack-trace@0.0.9, eyes@0.1.8, colors@0.6.2, pkginfo@0.3.1, request@2.16.6)

coveralls@2.11.6 node_modules\coveralls
├── lcov-parse@0.0.6
├── log-driver@1.2.4
├── minimist@1.2.0
├── js-yaml@3.0.1 (esprima@1.0.4, argparse@0.1.16)
└── request@2.67.0 (forever-agent@0.6.1, aws-sign2@0.6.0, tunnel-agent@0.4.2, is-typedarray@1.0.0, oauth-sign@0.8.0, form-data@1.0.0-rc3, caseless@0.11.0, stringstream@0.0.5, isstream@0.1.2, json-stringify-safe@5.0.1, extend@3.0.0, tough-cookie@2.2.1, qs@5.2.0, node-uuid@1.4.7, mime-types@2.1.8, combined-stream@1.0.5, bl@1.0.0, hawk@3.1.2, http-signature@1.1.0, har-validator@2.0.3)

chokidar@1.1.0 node_modules\chokidar
├── path-is-absolute@1.0.0
├── anymatch@1.3.0
├── arrify@1.0.1
├── glob-parent@2.0.0
├── async-each@0.1.6
├── is-glob@2.0.1 (is-extglob@1.0.0)
├── is-binary-path@1.0.1 (binary-extensions@1.4.0)
└── readdirp@2.0.0 (graceful-fs@4.1.2, readable-stream@2.0.5, minimatch@2.0.10)

jshint@2.8.0 node_modules\jshint
├── strip-json-comments@1.0.4
├── exit@0.1.2
├── shelljs@0.3.0
├── lodash@3.7.0
├── console-browserify@1.1.0 (date-now@0.1.4)
├── cli@0.6.6 (glob@3.2.11)
├── htmlparser2@3.8.3 (domelementtype@1.3.0, entities@1.0.0, domhandler@2.3.0, readable-stream@1.1.13, domutils@1.5.1)
└── minimatch@2.0.10 (brace-expansion@1.1.2)

istanbul@0.3.22 node_modules\istanbul
├── abbrev@1.0.7
├── wordwrap@1.0.0
├── nopt@3.0.6
├── esprima@2.5.0
├── once@1.3.3 (wrappy@1.0.1)
├── supports-color@3.1.2 (has-flag@1.0.0)
├── mkdirp@0.5.1 (minimist@0.0.8)
├── which@1.2.1 (is-absolute@0.1.7)
├── escodegen@1.7.1 (estraverse@1.9.3, esutils@2.0.2, esprima@1.2.5, optionator@0.5.0, source-map@0.2.0)
├── js-yaml@3.4.6 (inherit@2.2.2, esprima@2.7.1, argparse@1.0.3)
├── handlebars@4.0.5 (source-map@0.4.4, optimist@0.6.1, uglify-js@2.6.1)
└── fileset@0.2.1 (minimatch@2.0.10)

fs-extra@0.24.0 node_modules\fs-extra
├── path-is-absolute@1.0.0
├── jsonfile@2.2.3
├── graceful-fs@4.1.2
└── rimraf@2.5.0 (glob@6.0.2)

grunt-concurrent@2.0.4 node_modules\grunt-concurrent
├── indent-string@2.1.0 (repeating@2.0.0)
└── pad-stream@1.2.0 (split2@1.1.0, repeating@2.0.0, through2@2.0.0, pumpify@1.3.3, meow@3.6.0)

conventional-changelog@0.4.3 node_modules\conventional-changelog
├── q@1.4.1
├── add-stream@1.0.0
├── git-semver-tags@1.1.0
├── lodash@3.10.1
├── compare-func@1.3.1 (array-ify@1.0.0, dot-prop@2.2.0)
├── tempfile@1.1.1 (os-tmpdir@1.0.1)
├── dateformat@1.0.12 (get-stdin@4.0.1)
├── through2@2.0.0 (xtend@4.0.1, readable-stream@2.0.5)
├── conventional-commits-parser@0.1.2 (trim-off-newlines@1.0.0, is-text-path@1.0.1, split@1.0.0, JSONStream@1.0.7)
├── git-raw-commits@0.1.2 (split2@1.1.0, dargs@4.0.1, lodash.template@3.6.2)
├── conventional-changelog-writer@0.3.2 (conventional-commits-filter@0.1.1, split@1.0.0, handlebars@3.0.3)
├── get-pkg-repo@0.1.0 (hosted-git-info@2.1.4, normalize-package-data@2.3.5)
└── meow@3.6.0 (trim-newlines@1.0.0, object-assign@4.0.1, minimist@1.2.0, camelcase-keys@2.0.0, loud-rejection@1.2.0, normalize-package-data@2.3.5, redent@1.0.0, read-pkg-up@1.0.1)

vinyl-fs@2.0.0 node_modules\vinyl-fs
├── is-valid-glob@0.3.0
├── object-assign@4.0.1
├── graceful-fs@4.1.2
├── strip-bom@2.0.0 (is-utf8@0.2.1)
├── strip-bom-stream@1.0.0 (first-chunk-stream@1.0.0)
├── through2-filter@2.0.0 (xtend@4.0.1)
├── gulp-sourcemaps@1.6.0 (convert-source-map@1.1.2)
├── vinyl@1.1.0 (clone-stats@0.0.1, clone@1.0.2, replace-ext@0.0.1)
├── merge-stream@1.0.0 (readable-stream@2.0.5)
├── mkdirp@0.5.1 (minimist@0.0.8)
├── through2@2.0.0 (xtend@4.0.1, readable-stream@2.0.5)
├── duplexify@3.4.2 (readable-stream@2.0.5, end-of-stream@1.0.0)
└── glob-stream@5.3.1 (unique-stream@2.2.0, extend@3.0.0, to-absolute-glob@0.1.1, glob-parent@2.0.0, through2@0.6.5, ordered-read-streams@0.3.0, micromatch@2.3.7)

```

Then I run `socketstream new ss-app2` and I get

```
socketstream new ss-app2
Success! Created app 'ss-app2' with:
 ✓ Basic chat demo (-m for minimal install)
 ✓ Javascript example code (-c if you prefer CoffeeScript)
 ✓ Plain HTML for views (-j if you prefer Jade)
 ✓ Plain CSS (-s if you prefer Stylus, -l for Less)
Next, run the following commands:
   cd ss-app2
   [sudo] npm install
To start your app:
   node app.js
```
So, I follow the instructions and I cd to the ss-app2 folder, and run `npm install`. Then I open `http://localhost:3000/` and I get the sample app. I find some issues with the app though, but that is a different [work in progress](https://github.com/socketstream/socketstream/commit/c6d357d2db6f981625a5efe09cdadd8a0533208d) I suppose.

So, I believe the second part of the issue has been addressed by #603, but some [devDependencies](https://github.com/socketstream/socketstream/blob/develop/package.json#L49) are still not installed.

Regards

--- a/package.json
+++ b/package.json
@@ -41,12 +41,12 @@
     "shortid": "2.2.2",
     "uglify-js": "2.4.24",
     "utils-merge": "1.0.0",
-    "uuid": "2.0.1"
+    "uuid": "2.0.1",
+    "commander": "2.8.1"
   },
   "devDependencies": {
     "chai": "^3.3.0",
     "chokidar": "~1.1.0",
-    "commander": "2.8.1",
     "connect-livereload": "~0.5.2",
     "conventional-changelog": "^0.4.3",
     "cookie-parser": "^1.4.0",

